src/error.rs

```diff
@@ -8,14 +8,56 @@ pub enum ASEError {
     /// An error was encountered while parsing the ASE.
     ///
     /// This means that the input data did not conform to the ASE specification.
-    Invalid,
+    Invalid(ConformationError),
+    /// An error occured due to an invalid color format.
+    ///
+    /// Valid color formats are: CMYK, RGB, LAB, Gray.
+    ColorFormat,
+    /// An error occured due to Utf16 parsing issues.
+    UTF16Error,
+    /// An error occured due to an invalid color type.
+    ColorTypeError,
+    /// An error occured due to an invalid block type.
+    BlockTypeError,
+    /// An error occured while parsing the input data.
+    InputDataParseError,
+}
+
+/// Indicates the cause of the file being an invalid ase.
+#[derive(Debug)]
+pub enum ConformationError {
+    /// An invalid file version was detected.
+    ///
+    /// Currently on a version of 1.0 is supported.
+    FileVersion,
+    /// An invalid file signature was detected.
+    ///
+    /// The file signature must be 'ASEF'.
+    FileSignature,
```

Comment:
```suggestion
    /// An invalid file signature was detected.
    ///
    /// The file signature must be `ASEF`.
    FileSignature,
```
Improves formatting in rust-doc